### ISSUE TITLE
Dependencies: Adopting trust-dns-server@0.15

### ISSUE THREAD

**Issue by hdhoang** [valeriansaliou/constellation#5]
Hello! 

As part of the effort to merge rust-lang/rust#49799, trust-dns-server has bumped its rusqlite dependency.

I would like to create this tracking issue about porting constellation to 0.15 so that we will be ready once that PR lands.

**Comment by valeriansaliou** [comment]
Thank you, I'll batch-process this with other Constellation issues soon.

**Comment by Centril** [comment]
Any progress? :)

**Comment by valeriansaliou** [comment]
Hi! Thanks for pinging me back! Migrating this over the weekend. :)

--

Valerian Saliou
CTO at Crisp

**Comment by valeriansaliou** [comment]
Unfortunately, `trust-dns@0.15` and `trust-dns-server@0.15` use a different version of `trust-dns-proto`, which makes both libraries unusable when used at the same time (which is my case in Constellation; I need them both — compilation fails with a conflict error).

Attempting to upgrade to `0.14` at the moment; I guess `0.15` is blocking for now.

**Comment by valeriansaliou** [comment]
BTW, as can be seen on: https://deps.rs/repo/github/valeriansaliou/constellation ; I've managed to bump all other deps smoothly. Still attempting to bump all `trust-dns` deps.

**Comment by drahnr** [comment]
Do you have a working branch for this?

**Comment by valeriansaliou** [comment]
No, not yet. Happy to accept any PR in that direction though!

**Comment by drahnr** [comment]
Had a brief look into it, but the trust-dns project got restructured quite heavily, so it's hard to tell how much work it's going to be. From a compile perspective it seems to be mostly missing associative types. Let's see, I'll give it a whirl in the next few weeks.

**Comment by valeriansaliou** [comment]
Did a quick migration to trust-dns 0.14 in this PR: https://github.com/valeriansaliou/constellation/pull/21

**Comment by valeriansaliou** [comment]
Closing this in favor of PR #27
